File: iosApp/iosApp/views/screens/favorites/FavoritesScreenEvents.swift

import SwiftUI
import Shared


enum FavoritesUiEvent {
    case onItemClicked(FavoriteListItem)
    case removeFavorite(String)
}

struct FavoritesEventHandler {
    let events: Events
    let navActions: NavigationActions
    
    func onEvent(_ event: FavoritesUiEvent) {
        switch event {
        case .onItemClicked(let item):
            navActions.toDetailFromList(item.toDetailsNavParams())
        case .removeFavorite(let id):
            events.removeFromFavorites(code: id)
        }
    }
}

extension FavoriteListItem {
    func toDetailsNavParams() -> DetailsNavParams {
        DetailsNavParams(id: id, name: name)
    }
}

File: composeApp/src/androidMain/kotlin/com/exploramus/app/composables/screens/favorites/FavoritesScreenEvents.kt
package com.exploramus.app.composables.screens.favorites

import com.exploramus.app.composables.navigation.controller.DetailsNavParams
import com.exploramus.app.composables.navigation.controller.FavoritesPagerNavParams
import com.exploramus.app.composables.navigation.controller.ScreenNavActions
import com.exploramus.shared.viewmodel.core.Events
import com.exploramus.shared.viewmodel.screens.favorites.FavoriteListItem
import com.exploramus.shared.viewmodel.screens.favorites.removeFromFavorites

sealed class FavoritesUiEvent {
    data class RemoveFavorite(val countryId: String) : FavoritesUiEvent()
    data class OnItemClicked(val item: FavoriteListItem) : FavoritesUiEvent()
}

class FavoritesEventHandler(
    val navActions: ScreenNavActions,
    val events: Events
) {
    fun onEvent(event: FavoritesUiEvent) {
        when (event) {
            is FavoritesUiEvent.OnItemClicked -> {
                navActions.toFavoritesPagerFromList(event.item.toPagerNavParams())
            }
            is FavoritesUiEvent.RemoveFavorite -> events.removeFromFavorites(event.countryId)
        }
    }
}

fun FavoriteListItem.toDetailsNavParams() = DetailsNavParams(id, name)
fun FavoriteListItem.toPagerNavParams() = FavoritesPagerNavParams(id, name)

